src/parser/shadowlands/modules/items/raid/sepulcherofthefirstones/EarthbreakersImpact.tsx

```diff
@@ -0,0 +1,329 @@
+
+  private sessions: Session[] = [];
+  private currentSession: Session = {};
+
+  constructor(
+    options: Options & {
+      abilities: Abilities;
+      buffs: Buffs;
+      castEfficiency: CastEfficiency;
+    },
+  ) {
+    super(options);
+
+    this.item = this.selectedCombatant.getItem(ITEMS.EARTHBREAKERS_IMPACT.id)!;
+    if (this.item == null) {
+      this.active = false;
+      return;
+    }
+
+    // Add the buff to the buff tracker so that they show up in the timeline
+    options.buffs.add({
+      spellId: BUFF.id,
+      timelineHighlight: true,
+      triggeredBySpellId: CAST.id,
+    });
+
+    // Add the cast as an ability to track cooldown and usage in timeline
+    options.abilities.add({
+      spell: CAST.id,
+      category: Abilities.SPELL_CATEGORIES.ITEMS,
+      cooldown: COOLDOWN_SECONDS,
+      gcd: null,
+      castEfficiency: {
+        suggestion: true,
+        recommendedEfficiency: 0.8,
+      },
+    });
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(CAST), this.onCast);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(TICK), this.onDamage);
+  }
+
+  onCast(event: CastEvent) {
+    this.currentSession = {};
+    this.sessions.push(this.currentSession);
+  }
+
+  onDamage(event: DamageEvent) {
+    this.trackDamage(
+      event.timestamp,
+      event.targetID,
+      event.targetInstance,
+      event.amount,
+      event.hitType,
+    );
+  }
+
+  private trackDamage(
+    timestamp: number,
+    targetId: number,
+    targetInstance: number,
+    damage: number,
+    hitType: number,
+  ): void {
+    if (!this.currentSession[timestamp]) {
+      // There is no entry for this timestamp
+      this.currentSession[timestamp] = [];
+    } else if (
+      this.currentSession[timestamp].some(
+        (t) => t.targetId === targetId && t.targetInstance === targetInstance,
+      )
+    ) {
+      // The current target has already been hit by this tick
+      // so we try to create a new tick for the next millisecond to create separate ticks
+      // You might think that this would be so rare that it should not be necessary to handle
+      // but with the 2 example logs I looked at, one had this happen, so I assume it can happen
+      // again.
+      return this.trackDamage(timestamp + 1, targetId, targetInstance, damage, hitType);
+    }
+
+    this.currentSession[timestamp].push({
+      targetId,
+      targetInstance,
+      damage,
+      hitType,
+    });
+  }
+
+  private getCastEfficiency() {
+    return this.castEfficiency.getCastEfficiencyForSpellId(CAST.id);
+  }
+
+  getStats() {
+    const stats = this.sessions
+      // Summarize stats for each session
+      .map((session) => {
+        const timestamps = Object.keys(session).map(Number);
+
+        const tickTimestamps = findPeroidicNumbers(timestamps, TICK_MS);
+        const weakPointTimestamps = timestamps.filter(
+          (timestamp) => !tickTimestamps.includes(timestamp),
+        );
```

Comment:
so if i'm understanding this right, the reason this is done is because the weak points are the same spell id as the regular ticks? and you want to separate them?

Reply:
Yeah exactly. It's all damage with the same id. So I try to find the timestamps that is every 2 seconds so that I can identify the weak points.